ash-window/src/lib.rs

```diff
@@ -33,6 +33,47 @@ use raw_window_handle::{RawDisplayHandle, RawWindowHandle};
 ///
 /// [parent/child relation]: https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#fundamentals-objectmodel-lifetime
 pub unsafe fn create_surface(
+    entry: &Entry,
+    instance: &Instance,
+    display_handle: DisplayHandle<'_>,
+    window_handle: WindowHandle<'_>,
+    allocation_callbacks: Option<&vk::AllocationCallbacks>,
+) -> VkResult<vk::SurfaceKHR> {
+    #[allow(deprecated)]
+    create_surface_raw(
+        entry,
+        instance,
+        display_handle.as_raw(),
+        window_handle.as_raw(),
+        allocation_callbacks,
+    )
+}
+
+/// Create a surface from a raw surface handle.
+///
+/// `instance` must have created with platform specific surface extensions enabled, acquired
+/// through [`enumerate_required_extensions()`].
+///
+/// # Safety
+///
+/// There is a [parent/child relation] between [`Instance`] and [`Entry`], and the resulting
+/// [`vk::SurfaceKHR`].  The application must not [destroy][Instance::destroy_instance()] these
+/// parent objects before first [destroying][khr::Surface::destroy_surface()] the returned
+/// [`vk::SurfaceKHR`] child object.  [`vk::SurfaceKHR`] does _not_ implement [drop][drop()]
+/// semantics and can only be destroyed via [`destroy_surface()`][khr::Surface::destroy_surface()].
+///
+/// See the [`Entry::create_instance()`] documentation for more destruction ordering rules on
+/// [`Instance`].
+///
+/// The window represented by `window_handle` must be associated with the display connection
+/// in `display_handle`.
+///
+/// `window_handle` and `display_handle` must be associated with a valid window and display
+/// connection, which must not be destroyed for the lifetime of the returned [`vk::SurfaceKHR`].
+///
+/// [parent/child relation]: https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#fundamentals-objectmodel-lifetime
+#[deprecated = "Direct use of raw window handles is deprecated"]
```

Comment:
Is there a use case for keeping this at all? Users can already go from raw -> non-raw in unsafe code if an escape hatch is needed.

Reply:
Sure, there's the easy-to-find `borrow_raw()` helper function.

Inversely, since the returned `ash::SurfaceKHR` type has no notion of an implicit lifetime on this type, we could exclusively keep the raw function, and rely on the user to call the safe `as_raw()` function?

Reply:
True. I think semantically that's slightly cleaner, since we can't do anything with the lifetime information. On the other hand, windowing crates will presumably tend to emit non-raw handles (since `HasRaw...` are deprecated upstream), which makes that form slightly more concise to use.

I lean towards keeping the raw one only, since it's harder to miss the safety invariants that way. So long as we're only exposing one form I'm happy regardless, though.

Reply:
Let's stick with the `Raw` interface for those reasons, until upstream starts pestering that code-path with `#[deprecated]` :)

Reply:
> On the other hand, windowing crates will presumably tend to emit non-raw handles (since `HasRaw...` are deprecated upstream)

Fortunately we're not taking any of the `Has*` trait variants, so this should be less of an issue besides the user calling `event_loop.display_handle().as_raw()`/`window.window_handle().as_raw()` now.

Reply:
Yeah, I think that aligns well with upstream's intent.